scripts/maint/pre-push.git-hook

```diff
@@ -1,17 +1,32 @@
 #!/bin/bash
 
+# git pre-push hook script to prevent "fixup!" and "squash!" commit
+# from ending up in master, or in any branch if CUR_BRANCH check is removed.
+# Furthermore, it disallows pushing branches other than master, release-*
+# and maint-* to origin (e.g. gitweb.torproject.org).
+#
 # To install this script, copy it into .git/hooks/pre-push path in your
 # local copy of git repository. Make sure it has permission to execute.
 #
-# This is git pre-push hook script to prevent "fixup!" and "squash!" commits
-# from ending up in upstream branches (master, release-* or maint-*).
-#
 # The following sample script was used as starting point:
 # https://github.com/git/git/blob/master/templates/hooks--pre-push.sample
 
 z40=0000000000000000000000000000000000000000
 
+remote="$1"
 CUR_BRANCH=$(git rev-parse --abbrev-ref HEAD)
```

Comment:
I know this line isn't new, but does it still work if you're pushing something other than HEAD, as in "git push branchname reponame" ?

Reply:
Good point. I tried to make it more general in 60c8af367799e2cb8f59a236e97fdb0ad04bc350.